packages/styled-system/package.json

```diff
@@ -45,7 +45,8 @@
   },
   "sideEffects": false,
   "scripts": {
-    "build": "tsup src/index.ts --dts",
+    "build": "tsup src/index.ts && pnpm build:dts",
+    "build:dts": "tsup --entry src/index.ts --entry src/theming.types.ts --entry src/shared.types.ts --dts-only",
```

Comment:
How about we update the the `build` command to this:

```sh
tsup src/ --dts
```

That way, it'll generate individual chunks for the code and types and everything works as needed.

Reply:
I tested `tsup src/ --dts` and this command generated some duplicated code.
So I think `tsup src/index.ts` is more efficient and better.